### PR TITLE
CB-7085 - Env: moving UMS api calls from nio thread to flow

### PR DESCRIPTION
**Pull request by attilapalfi92**

### CODE REVIEW

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentInitHandler.java [hunk 1]
```diff
@@ -52,6 +62,34 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
...
+        String environmentCrnForVirtualGroups = environment.getResourceCrn();
+        if (Objects.nonNull(environment.getParentEnvironment())) {
+            environmentCrnForVirtualGroups = environment.getParentEnvironment().getResourceCrn();
+        }
+        return environmentCrnForVirtualGroups;
+    }
+
+    private boolean createVirtualGroups(Environment environment, String envCrn) {
+        boolean result = false;
+        if (StringUtils.isEmpty(environment.getAdminGroupName())) {
```
Comment (doktoric) [inline environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentInitHandler.java:85]: isNullorEmpty ?

Comment (bergerdenes) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentInitHandler.java:85]: isblank? :D

Comment (attilapalfi92) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentInitHandler.java:85]: dont worry they all do the same. :D I like the isEmpty named version

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentInitHandler.java [hunk 2]
```diff
@@ -52,6 +62,34 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
...
+            environmentService.setAdminGroupName(environment, environment.getAdminGroupName());
+        }
+        environmentService.assignEnvironmentAdminAndOwnerRole(environment.getCreator(), environmentCrnForVirtualGroups);
+        setLocationAndRegions(environment);
+    }
+
+    private String getEnvironmentCrnForVirtualGroups(Environment environment) {
+        String environmentCrnForVirtualGroups = environment.getResourceCrn();
+        if (Objects.nonNull(environment.getParentEnvironment())) {
+            environmentCrnForVirtualGroups = environment.getParentEnvironment().getResourceCrn();
```
[the remark this answers is not in the document]
Comment (attilapalfi92) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentInitHandler.java:78]: It is not an api call or anything, returns a reference. I can cache that reference but I think the JVM will do that.

[the remark this answers is not in the document]
Comment (attilapalfi92) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/EnvironmentInitHandler.java:78]: (I moved this code without modification from the env creation service)

**Pull request merged**